Purchasing.Mvc/Attributes/IpFilter.cs

```diff
@@ -1,24 +1,25 @@
 ﻿using System;
 using System.Configuration;
 using System.Net.Mail;
-using System.Web.Mvc;
-using System.Web.Routing;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Configuration;
+using UCDArch.Core;
 
 namespace Purchasing.Mvc.Attributes
 {
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false)]
-    public class IpFilter : FilterAttribute, IAuthorizationFilter
+    public class IpFilter : Attribute, IAuthorizationFilter
     {
-        private string localIpFilter = ConfigurationManager.AppSettings["IpFilter"];
-
-        public void OnAuthorization(AuthorizationContext filterContext)
+        public void OnAuthorization(AuthorizationFilterContext filterContext)
         {
-            var clientIp = filterContext.RequestContext.HttpContext.Request.UserHostAddress;
+            var localIpFilter = SmartServiceLocator<IConfiguration>.GetService()["IpFilter"];
+            var clientIp = filterContext.HttpContext.Connection.RemoteIpAddress.MapToIPv4().ToString();
 
             // not authorized
             if (clientIp != localIpFilter)
             {
-                filterContext.Result = new RedirectToRouteResult(new RouteValueDictionary(new { controller = "error", action = "notauthorized" }));
+                filterContext.Result = new UnauthorizedObjectResult("You do not have access to this application");
```

Comment:
This isn't really the same, but it doesn't look like we use this either, so no worries.